Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm

```diff
@@ -12317,7 +12317,21 @@ - (void)continueContextMenuInteraction:(void(^)(UIContextMenuConfiguration *))co
 
 #if ENABLE(DATA_DETECTION)
     if ([(NSURL *)linkURL iTunesStoreURL]) {
-        [self continueContextMenuInteractionWithDataDetectors:continueWithContextMenuConfiguration];
+        bool shouldContinueContextMenuInteraction = [&] {
+            if (!_page->websiteDataStore().isPersistent())
+                return false;
+
+#if ENABLE(NETWORK_CONNECTION_INTEGRITY)
+            if (_page->networkConnectionIntegrityPolicies().contains(WebCore::NetworkConnectionIntegrity::Enabled))
+                return false;
+#endif
+
+            return true;
+        }();
```

Comment:
(optional) Nit: since we're only using `_page` maybe just capture that?

```suggestion
        bool shouldContinueContextMenuInteraction = [&page = _page] {
            if (!page->websiteDataStore().isPersistent())
                return false;

#if ENABLE(NETWORK_CONNECTION_INTEGRITY)
            if (page->networkConnectionIntegrityPolicies().contains(WebCore::NetworkConnectionIntegrity::Enabled))
                return false;
#endif

            return true;
        }();
```

Reply:
I think either approach seems reasonable. This check also seems simple enough, that you could consider removing the temporary variable altogether and just do:

```
if (_page->websiteDataStore().isPersistent() && !_page->networkConnectionIntegrityPolicies().contains(WebCore::NetworkConnectionIntegrity::Enabled))
    [self continueContextMenuInteractionWithDataDetectors:continueWithContextMenuConfiguration];
else
    continueWithContextMenuConfiguration(nil);
return;
```

Reply:
@whsieh what about the compile time guards though?

Reply:
Oh, right. Something like this?

```
    bool shouldUseDataDetectors = _page->websiteDataStore().isPersistent();
#if ENABLE(NETWORK_CONNECTION_INTEGRITY)
    if (_page->networkConnectionIntegrityPolicies().contains(WebCore::NetworkConnectionIntegrity::Enabled))
        shouldUseDataDetectors = false;
#endif
    …
```